### PR TITLE
Add Thread Certification 9.2.4 to automated test.

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: src/cli/cli_dataset.cpp [hunk 1]
```diff
@@ -527,6 +527,13 @@ ThreadError Dataset::ProcessMgmtSetCommand(otInstance *aInstance, int argc, char
             SuccessOrExit(error = Interpreter::ParseLong(argv[++index], value));
             dataset.mChannel = static_cast<uint16_t>(value);
         }
+        else if (strcmp(argv[index], "channelmask") == 0)
```
Comment (xiaom-GitHub) [inline src/cli/cli_dataset.cpp]: Should we also need to append channel mask tlv in `SendSetRequest()` if this tlv is set by CLI?

Comment (jwhui) [inline src/cli/cli_dataset.cpp]: Yes, good catch.  I updated this PR to add that.  Thanks.

File: src/core/thread/meshcop_tlvs.hpp [hunk 2]
```diff
@@ -1358,12 +1358,31 @@ class ChannelMask0Tlv: public ChannelMaskTlv, public ChannelMaskEntry
...
-    uint32_t GetMask(void) { return HostSwap32(mMask); }
+    uint32_t GetMask(void) { return Thread::Encoding::LittleEndian::HostSwap32(mMask); }
+
+    /**
+     * This method sets the Channel Mask value.
+     *
+     * @param[in]  aMask  The Channel Mask value.
+     *
+     */
+    void SetMask(uint32_t aMask) { mMask = Thread::Encoding::LittleEndian::HostSwap32(aMask); }
```
Comment (xiaom-GitHub) [inline src/core/thread/meshcop_tlvs.hpp]: 👍

**Comment by xiaom-GitHub** [comment]
LGTM 👍

**Pull request merged**